kalite/main/migrations/0039_log_timestamps.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        """
+        Create latest_activity_timestamp data when possible by inspecting the model.
+        For ExerciseLog inspect relevant AttemptLogs to infer most recent activity.
+        If not available, and for other Logs, fallback to completion_timestamp if available.
+        """
+        for e in orm.ExerciseLog.objects.filter(latest_activity_timestamp=None):
+
+            alogs = orm.AttemptLog.objects.filter(user=e.user, exercise_id=e.exercise_id, context_type__in=["playlist", "exercise"])
+
+            if alogs:
+                e.latest_activity_timestamp = sorted([a.timestamp for a in alogs])[-1]
+                e.zone_fallback = e.get_zone()
+                e.save()
+            else:
+                e.latest_activity_timestamp = e.completion_timestamp
+                e.save()
+
+        for v in orm.VideoLog.objects.filter(latest_activity_timestamp=None):
```

Comment:
For this one (and the ContentLog), the filter could be `.filter(latest_activity_timestamp=None).exclude(completion_timestamp=None)`, to avoid unnecessary saves.